libutils/regex.c

```diff
@@ -238,10 +218,84 @@ Seq *StringMatchCaptures(const char *regex, const char *str, const bool return_n
     }
 
     free(ovector);
+    return ret;
+}
+
+// Returns a Sequence with Buffer elements.
+
+// If return_names is set, the even positions will be the name or
+// number of the capturing group, followed by the captured data in the
+// odd positions (so for N captures you can expect 2N elements in the
+// Sequence).
+
+// If return_names is not set, only the captured data is returned (so
+// for N captures you can expect N elements in the Sequence).
+
+Seq *StringMatchCaptures(const char *regex, const char *str, const bool return_names)
+{
+    assert(regex);
+    assert(str);
+
+    pcre *pattern = NULL;
+    {
+        const char *errorstr;
+        int erroffset;
+        pattern = pcre_compile(regex, PCRE_MULTILINE | PCRE_DOTALL,
+                               &errorstr, &erroffset, NULL);
+    }
+
+    if (pattern == NULL)
+    {
+        return NULL;
+    }
+
+    Seq *ret = StringMatchCapturesWithPrecompiledRegex(pattern, str, return_names);
     pcre_free(pattern);
     return ret;
 }
 
+/*******************************************************************/
+
+// returns NULL on any failure
+// takes either a pre-compiled pattern OR a regex (one of the two shouldn't be NULL)
+JsonElement* StringCaptureData(pcre *pattern, const char* regex, const char* data)
```

Comment:
Unit tests fail because this function references `Json*` functions, and the test is built standalone, with only string and regex functions compiled in. Perhaps this should be moved to `json.c` and `json.h` instead?